projects/jobs/jobs.groovy

```diff
@@ -23,146 +32,186 @@ loadCartridgeJob.with{
     }
     environmentVariables {
+        groovy("return [SCM_KEY: org.apache.commons.lang.RandomStringUtils.randomAlphanumeric(20)]")
         env('WORKSPACE_NAME',workspaceFolderName)
         env('PROJECT_NAME',projectFolderName)
+        keepBuildVariables(true)
+        keepSystemVariables(true)
+        overrideBuildParameters(true)
     }
     wrappers {
         preBuildCleanup()
         injectPasswords()
         maskPasswords()
-        sshAgent("adop-jenkins-master")
+        credentialsBinding {
+            file('SCM_SSH_KEY', 'adop-jenkins-private')
+        }        
+        copyToSlaveBuildWrapper {
+          includes("**/**")
+          excludes("")
+          flatten(false)
+          includeAntExcludes(false)
+          relativeTo('''${JENKINS_HOME}/userContent''')
+          hudsonHomeRelative(false)
+        }
     }
+    label("!master && !windows && !ios")
     steps {
         shell('''#!/bin/bash -ex
 
+mkdir ${WORKSPACE}/tmp
+
+# Output SCM provider ID to a properties file
+echo SCM_PROVIDER_ID=$(echo ${SCM_PROVIDER} | cut -d "(" -f2 | cut -d ")" -f1) > ${WORKSPACE}/scm.properties
+
+# Check if SCM namespace is specified
+if [ -z ${SCM_NAMESPACE} ] ; then
+    echo "SCM_NAMESPACE not specified, setting to PROJECT_NAME..."
+    if [ -z ${CARTRIDGE_FOLDER} ] ; then
+      SCM_NAMESPACE="${PROJECT_NAME}"
+    else
+      SCM_NAMESPACE="${PROJECT_NAME}"/"${CARTRIDGE_FOLDER}"
+    fi
+else
+    echo "SCM_NAMESPACE specified, injecting into properties file..."
+fi
+
+echo SCM_NAMESPACE=$(echo ${SCM_NAMESPACE} | cut -d "(" -f2 | cut -d ")" -f1) >> ${WORKSPACE}/scm.properties
+''')
+        environmentVariables {
+            propertiesFile('${WORKSPACE}/scm.properties')
+        }
+        systemGroovyCommand('''
+import com.cloudbees.plugins.credentials.*;
+import com.cloudbees.plugins.credentials.common.*;
+import pluggable.scm.PropertiesSCMProviderDataStore;
+import pluggable.scm.SCMProviderDataStore;
+import pluggable.configuration.EnvVarProperty;
+import pluggable.scm.helpers.PropertyUtils;
+import java.util.Properties;
+import hudson.FilePath;
+
+String scmProviderId = build.getEnvironment(listener).get('SCM_PROVIDER_ID')
+EnvVarProperty envVarProperty = EnvVarProperty.getInstance();
+
+envVarProperty.setVariableBindings(build.getEnvironment(listener));
+SCMProviderDataStore scmProviderDataStore = new PropertiesSCMProviderDataStore();
+Properties scmProviderProperties = scmProviderDataStore.get(scmProviderId);
+
+String credentialId = scmProviderProperties.get("loader.credentialId")
+
+if(credentialId != null){
+
+  def username_matcher = CredentialsMatchers.withId(credentialId);
+
+  def available_credentials = CredentialsProvider.lookupCredentials(
+    StandardUsernameCredentials.class
+  );
+
+  credentialInfo =  [CredentialsMatchers.firstOrNull(available_credentials, username_matcher).username,
+                  CredentialsMatchers.firstOrNull(available_credentials, username_matcher).password];
+
```

Comment:
This will throw a null point exception if the credential is not found.
java.lang.NullPointerException: Cannot get property 'username' on null object".

Please can we wrap the username/password extraction so if it throws a null a helpful warning message is displayed to the user.